nx/lib/nx/serving.ex

```diff
@@ -1300,7 +1323,8 @@ defmodule Nx.Serving do
       }
     )
 
-    :pg.join(Nx.Serving.PG, __MODULE__, List.duplicate(self(), partitions_count))
+    serving_weight = max(1, weight) * partitions_count
```

Comment:
In case we want to support floats in the future, this is the sanest default, right?

```suggestion
    serving_weight = max(1, weight * partitions_count)
```